static/js/topic_page/line_tile.tsx

```diff
@@ -0,0 +1,169 @@
+import { getStatsVarLabel } from "../shared/stats_var_labels";
+import { CHART_HEIGHT } from "./constants";
+import { StatVarMetadata } from "./types";
+
+const SVG_CONTAINER_ELEMENT: React.RefObject<HTMLDivElement> = React.createRef();
+
+interface LineTilePropType {
+  id: string;
+  title: string;
+  placeDcid: string;
+  statVarMetadata: StatVarMetadata;
+}
+
+export function LineTile(props: LineTilePropType): JSX.Element {
+  const [rawData, setRawData] = useState<StatApiResponse | undefined>(null);
+  const [chartData, setChartData] = useState<DataGroup[] | undefined>(null);
+
+  useEffect(() => {
+    fetchData(props.statVarMetadata);
+  }, []);
+
+  useEffect(() => {
+    if (rawData) {
+      processData(rawData);
+    }
+  });
+
+  useEffect(() => {
+    if (chartData) {
+      drawChart(chartData);
+    }
+  });
+
+  return (
+    <div className="chart-container">
+      {chartData && (
+        <>
+          <div className="line-title">
+            <h4>{props.title}</h4>
+          </div>
+          <div
+            id={props.id}
+            className="svg-container"
+            ref={SVG_CONTAINER_ELEMENT}
+          ></div>
+        </>
+      )}
+    </div>
+  );
+
+  function fetchData(statVarMetaData: StatVarMetadata): void {
+    axios
+      .post(`/api/stats`, {
+        // Fetch both numerator stat vars and denominator stat vars
+        statVars: statVarMetaData.statVars.concat(statVarMetaData.denominator),
+        places: [props.placeDcid],
+      })
+      .then((resp) => {
+        setRawData(resp.data);
+      })
+      .catch(() => {
+        // TODO: add error message
+        setRawData(null);
+      });
+  }
+
+  function processData(rawData: StatApiResponse): void {
+    const trendData = rawToChart(rawData);
+    setChartData(trendData);
+  }
+
+  function drawChart(chartData: DataGroup[]): void {
+    const elem = document.getElementById(props.id);
+    elem.innerHTML = "";
+    const isCompleteLine = drawLineChart(
+      props.id,
+      elem.offsetWidth,
+      CHART_HEIGHT,
+      chartData,
+      false,
+      false,
+      props.statVarMetadata.unit
+    );
+    if (!isCompleteLine) {
+      SVG_CONTAINER_ELEMENT.current.querySelectorAll(
+        ".dotted-warning"
+      )[0].className += " d-inline";
+    }
+  }
+
+  function rawToChart(rawData: StatApiResponse): DataGroup[] {
+    // (TODO): We assume the index of numerator and denominator matches.
+    // This is brittle and should be updated in the protobuf that binds both
+    // together.
+    const raw = _.cloneDeep(rawData);
+    const dataGroups: DataGroup[] = [];
+    const sources = new Set<string>();
+    const metadata = props.statVarMetadata;
+    const computeRatio =
+      metadata.statVars.length === metadata.denominator.length;
+    const allDates = new Set<string>();
+    for (let i = 0; i < metadata.statVars.length; i++) {
+      const numSV = metadata.statVars[i];
+      console.log(numSV);
```

Comment:
Probably remove this and the rest of the console.logs in this function?

Reply:
Done